ethereum/README.md

```diff
@@ -2,61 +2,101 @@
+    - ERC20App.sol: application for cross-chain ERC20 transfers between Ethereum and Substrate
+- Decoder.sol: a library for decoding SCALE encoded data
+- Verifier.sol: verifies tx origin and signatures
+- Scale.sol: implements decoding of SCALE encoded compact uints; not currently used, will support future work on generalized data relay
+
 ## Set up
 
-Create a `.env` file using the following template:
+After starting the blockchain and deploying the contracts, the Bridge's Ethereum component is ready for usage. To generate configuration information required for configuring the Parachain and Relayer components, run the set up scripts described in scripts section below.
 
 ```bash
-cp .env.example .env
+# Install dependencies
+yarn install
 ```
 
-Install dependencies
+Start truffle environment with a local Ethereum network and deploy the contracts
 
 ```bash
-yarn install
+truffle develop
+truffle migrate --all
 ```
 
-Start truffle environment containing a local Ethereum network
+## Testing
+
+Make sure the truffle environment is running, then run tests
 
 ```bash
+# Start testing environment if it's not already running
 truffle develop
+
+# Test application gas expenditures
+truffle test test/test_gas.js
+
+# Run all tests
+truffle test
 ```
 
-Contract compilation
+## Scripts
+
+To run the scripts, create an `.env` file using the example template.
 
 ```bash
-truffle compile
+cp scripts/.env.example scripts/.env
 ```
 
-Migrate contracts to network
+The project includes several scripts for Bridge interaction:
 
-```bash
-truffle migrate --reset --all
+`getTx.js` gets information about a transaction
+
+``` bash
+truffle exec scripts/getTx.js [tx-hash]
 ```
 
-## Testing
+`getEthBalance.js` gets an address' ETH balance
+
+``` bash
+truffle exec scripts/getEthBalance.js [ethereum-address]
+```
 
-Make sure the truffle environment is running
+`getERC20balance.js` gets an address' ERC20 token balance
 
-```bash
-truffle develop
+``` bash
+truffle exec scripts/getERC20Balance.js [ethereum-address] [token-contract-address]
 ```
 
-In another terminal, run tests
+`sendEth.js` deposits ETH into the ETHApp, initiating a cross-chain transfer to Substrate
 
-```bash
-truffle test
+``` bash
+truffle exec scripts/sendEth.js [amount] [polkadot-recipient]
 ```
 
-Test gas expenditures
+`sendERC20.js` deposits ERC20 tokens into the ERC20App, initiating a cross-chain transfer to Substrate
 
-```bash
-truffle test test/test_gas.js
+``` bash
+truffle exec scripts/sendERC20.js [amount] [token-contract-address] [polkadot-recipient]
 ```
 
-## Future Work
+Several additional scripts support the set up of other components:
+
+`dumpABI.js` outputs the ABI of the Bridge, ETHApp, and ERC20App contracts
 
-- Polkadot Transaction Verifier
-- Arbitrary State Parser
-- Arbitrary State Submission
-- Polkadot Signature Prover
+``` bash
+truffle exec scripts/dumpABI.js
+```
+
+`dumpParachainConfig.js`  outputs information required by the Parachain component
```

Comment:
I'd rather not document `dumpParachainConfig.js` and `dumpRelayerConfig.js`. They are just for internal build tooling and their interface is liable to change at any time.

Reply:
Removed in https://github.com/Snowfork/polkadot-ethereum/pull/107/commits/5d02b9d9f0e7f00a7a8082af1a854a096165dee6